Libraries/oneTBB/tbb-async-sycl/src/tbb-async-sycl.cpp

```diff
@@ -0,0 +1,195 @@
+//==============================================================
+// Copyright © 2019 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <cmath>  //for std::ceil
+#include <array>
+#include <atomic>
+#include <iostream>
+#include <thread>
+
+#include <CL/sycl.hpp>
+
+#include <tbb/blocked_range.h>
+#include <tbb/flow_graph.h>
+#include <tbb/global_control.h>
+#include <tbb/parallel_for.h>
+
+using namespace cl::sycl;
+
+constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
+constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+constexpr cl::sycl::access::mode sycl_read_write = cl::sycl::access::mode::read_write;
+
+struct done_tag{};
+
+// The exception_list parameter is an iterable list of std::exception_ptr objects.
+// But those pointers are not always directly readable.
+// So, we rethrow the pointer, catch it,  and then we have the exception itself.
+// Note: depending upon the operation there may be several exceptions.
+auto exception_handler = [](exception_list exceptionList) {
```

Comment:
Please remove the exception_handler here and add it as below when you instantiate the queue
queue q(default_selector{}, dpc_common::exception_handler);